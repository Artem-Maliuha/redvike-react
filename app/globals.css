@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --primary: #1A1A1E;
  --background-rgb: 255, 255, 255;
}

body {
  color: rgb(var(--primary));
  background: rgb(var(--background-rgb));
}
