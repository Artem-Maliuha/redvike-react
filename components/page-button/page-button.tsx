import Link from 'next/link'
import { ReactNode } from 'react'

interface PageButtonProps {
  href: string
  children: string | ReactNode
  active?: boolean
}
export const PageButton = (props: PageButtonProps) => {
  const { href, children, active = false } = props
  const basicClasses =
    'shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primary'
  const activeClasses = ' bg-primary text-white'
  return (
    <Link className={`${basicClasses}${active && activeClasses}`} href={href}>
      {children}
    </Link>
  )
}
