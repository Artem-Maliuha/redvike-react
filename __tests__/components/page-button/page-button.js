import renderer from 'react-test-renderer'
import { describe, it, expect } from '@jest/globals'
import { PageButton } from '../../../components/page-button'

describe('<PageButton />', () => {
  it('snapshot', () => {
    const component = renderer.create(<PageButton href="/">1</PageButton>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('snapshot active', () => {
    const component = renderer.create(
      <PageButton href="/" active>
        1
      </PageButton>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
