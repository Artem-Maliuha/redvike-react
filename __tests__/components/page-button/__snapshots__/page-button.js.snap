// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageButton /> snapshot 1`] = `
<a
  className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primaryfalse"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  1
</a>
`;

exports[`<PageButton /> snapshot active 1`] = `
<a
  className="shadow-hard flex size-[32px] items-center justify-center border-2 text-base font-bold border-primary bg-primary text-white"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  1
</a>
`;
